frontend/android/src/main/java/io/github/droidkaigi/confsched2019/App.kt

```diff
@@ -30,12 +33,17 @@ open class App : DaggerApplication() {
         setupFont()
         setupEmojiCompat()
         setupFirestore()
-        ProcessLifecycleOwner.get().lifecycle.addObserver(systemActionCreator)
-        systemStore.systemProperty.changedForever {
-            // listening
-            Timber.debug { "SystemProperty changed $it" }
-        }
-        systemActionCreator.setupSystem()
+        setupSystemActionCreator()
+    }
+
+    @OnLifecycleEvent(Lifecycle.Event.ON_CREATE)
+    private fun onProcessCreated() {
+        systemActionCreator.observeSystemProperty()
+    }
+
+    @OnLifecycleEvent(Lifecycle.Event.ON_DESTROY)
```

Comment:
Can you use ON_START ON_STOP?

> ON_DESTROY will never be dispatched

https://developer.android.com/reference/android/arch/lifecycle/ProcessLifecycleOwner

Reply:
If we use ON_START/ON_STOP, we cannot catch ACTION_LOCALE_CHANGED when app is background...
However, if ON_DESTROY is used, the broadcast receiver will not be unregistered. 🤔 

I implemented other version using BroadcastReceiver in AndroidManifest.xml, but this pattern got time-lagged until receive ACTION_LOCALE_CHANGED...

Reply:
Please let me think.🤔

Reply:
I guess re-checking SystemProperty in SessionPagesFragment is better than using dynamic broadcast receiver for fixing issue #187.
Something like this:

```kotlin
class SessionPagesFragment : DaggerFragment() {

override fun onStart() {
  ...
 systemActionCreator.checkSystemPropertyUpdated()
}
```

I don't know how to check current store state from SystemActionCreator...

Reply:
Well.. onStart in App, we can SystemActionCreator.refresh(). Now, We are using a changed extension function.
That function use `distinct()` in livedata-ktx. So probably it is OK.

Reply:
I added sample code 🙏 

```
open class App : DaggerApplication() {
...
        ProcessLifecycleOwner.get().lifecycle.addObserver(this)
...
    @OnLifecycleEvent(Lifecycle.Event.ON_START)
    private fun onStart() {
        systemActionCreator.refresh()
    }
```

Reply:
Thanks for the sample code.  😄 
I am working on it now, but It cannot reload as well. Please give me a little more time... 🙏

Reply:
I changed to use ON_START/ON_STOP, so broadcast receiver was unregistered when application go to background.
To detect ACTION_LOCALE_CHANGE while app is background, check lang when application come back to front.